cryptol-remote-api/python/README.md

```diff
@@ -1,83 +1,190 @@
-libraries to match the environment. For instance, `PATH` is modified
-to prioritize the virtual environment's Python version, and that
-Python is pointed at the specific collection of libraries that are
-available. Under a broadly Bourne-compatible shell like `bash` or
-`zsh`, source the appropriate file in the environment:
+First, clone the repository and submodules.
 
 ```
-$ . virtenv/bin/activate
+$ git clone https://github.com/GaloisInc/cryptol.git
+$ cd cryptol
+$ git submodule update --init
 ```
 
-to activate the environment. Deactivate it using the shell alias
-`deactivate` that is defined by `activate`. For other shells or
-operating systems, please consult the documentation for `venv`. If
-you prefer not to activate the environment, it is also possible to run
-the environment's version of Python tooling by invoking the scripts in
-its `bin` directory.
-
-In the virtual environment, run the following command to install the
-library's dependencies:
+Then, use [`poetry`](https://python-poetry.org/docs/#installation) to install
+the python client from the `cryptol-remote-api/python` directory:
 
 ```
-$ pip install -r requirements.txt
+$ cd cryptol-remote-api/python
+$ poetry install
 ```
 
-Next, install the library itself:
+# Cryptol server
+
+To run the verification scripts a `cryptol-remote-api` server must be available,
+either as a local executable or running in docker image listening on a port.
+
+## Connecting with a server in a script
+
+Connecting to a server in a Python script is accomplished via the `cryptol.connect`
+method. Its accompanying Python doc strings describe the various ways it can be
+used. Below is a brief summary:
+
+`cryptol.connect()`, when provided no arguments, will attempt the following in order:
+
+1. If the environment variable ``CRYPTOL_SERVER`` is set and refers to an
+   executable, it is assumed to be a `cryptol-remote-api` executable and will be
+   used for the duration of the script.
+2. If the environment variable ``CRYPTOL_SERVER_URL`` is set, it is assumed to be
+   the URL for a running Cryptol server in ``http`` mode and will be connected to.
+   (N.B., this can be a local server or a server running in a docker image.)
+3. If an executable ``cryptol-remote-api`` is available on the ``PATH`` it is
+    assumed to be a Cryptol server and will be used for the duration of the script.
+
+Additional arguments and options are documented with the function.
+
+Notably, the `reset_server` keyword can be used to connect to a running server
+and reset it, ensuring states from previous scripts have been cleared. E.g.,
+`cryptol.connect(reset_server=True)`.
+
+
+## Acquiring a Cryptol Server
+
+There are several ways a server executable can be obtained.
+
+### Server executables
+
+An executable of the server is now included in each release of Cryptol.
+
+Nightly server builds can be found under the 
+[Packages](https://github.com/orgs/GaloisInc/packages?repo_name=cryptol) section
```

Comment:
The packages here include only Docker images. The nightly builds are here: https://github.com/GaloisInc/cryptol/actions/workflows/nightly.yml

Reply:
Ah good point -- I've added pointers to both now.